wannierberri/__system.py

```diff
@@ -295,118 +295,112 @@ def cRvec(self):
 
     @lazy_property.LazyProperty
     def cRvec_wcc(self):
-        """ 
+        """
         With self.use_wcc_phase=True it is R+tj-ti. With self.use_wcc_phase=False it is R. [i,j,iRvec,a] (Cartesian)
         """
         if self.use_wcc_phase:
-            return self.iRvec.dot(self.real_lattice)[None,None,:,:]+ self.diff_wcc_cart[:,:,None,:]
+            return self.cRvec[None,None,:,:] + self.diff_wcc_cart[:,:,None,:]
         else:
-            return self.iRvec.dot(self.real_lattice)[None,None,:,:]
+            return self.cRvec[None,None,:,:]
 
 
     @lazy_property.LazyProperty
     def diff_wcc_cart(self):
-        """ 
+        """
         With self.use_wcc_phase=True it is tj-ti. With self.use_wcc_phase=False it is 0. [i,j,a] (Cartesian)
         """
         wannier_centers = self.wannier_centers_cart
-        return np.array([[j-i for j in wannier_centers] for i in wannier_centers])
+        return wannier_centers[None, :, :] - wannier_centers[:, None, :]
 
     @lazy_property.LazyProperty
     def diff_wcc_red(self):
-        """ 
-        With self.use_wcc_phase=True it is tj-ti. With self.use_wcc_phase=False it is 0. [m,n,a] (Reduced)
+        """
+        With self.use_wcc_phase=True it is tj-ti. With self.use_wcc_phase=False it is 0. [i,j,a] (Reduced)
         """
         wannier_centers = self.wannier_centers_reduced
-        return np.array([[j-i for j in wannier_centers] for i in wannier_centers])
+        return wannier_centers[None, :, :] - wannier_centers[:, None, :]
```

Comment:
btw, `diff_wcc_red` is not used anymore.  But we may leave it for future, just in case